restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java

```diff
@@ -803,28 +798,70 @@ void addRowsToMap(Map<String, List<Row>> rowsByDoc, List<Row> rows) {
     return Collections.emptyList();
   }
 
+  /**
+   * If the DocumentSearchPageState has a defined last seen `documentId`, advance the result set to
+   * be just after that ID.
+   */
+  private List<Row> skipSeenRows(DocumentSearchPageState initialPagingState, List<Row> rows) {
+    if (initialPagingState == null) {
+      return rows;
+    }
+
+    String lastSeenId = initialPagingState.getLastSeenDocId();
+    if (lastSeenId == null) {
+      return rows;
+    }
+
+    boolean idFound = false;
+    boolean doneSkipping = false;
+    while (!doneSkipping && rows.size() > 0) {
+      String docId = rows.get(0).getString("key");
+      if (docId.equals(lastSeenId)) {
+        idFound = true;
+        rows.remove(0);
+      } else if (idFound) {
+        doneSkipping = true;
+      } else {
+        rows.remove(0);
```

Comment:
I believe we also have to support the case when the reference ID is not found at all. Could this happen when data is deleted or inserted asynchronously?

Reply:
We may need to use something like `DecoratedKey` and `IPartitioner` to be able to detect when we've advanced past the previous ID.

Reply:
Ah that is an interesting edge-case...If the "last seen" ID was deleted, then some weird behavior will be observed and it will actually skip over the entire first page of results...And if data is inserted such that the "last seen" ID is pushed to the following page, the same will happen - so it will show the last seen ID's document again.

We discussed a bit about how inserts/deletions mid-stream would affect the final result, but yes let's talk through these edge cases (and how those two `DecoratedKey` and `IPartitioner` might help)